tuf/api/metadata.py

```diff
@@ -61,14 +62,25 @@
     SignedSerializer,
 )
 
+
+class RoleNames(Enum):
+    ROOT: str = "root"
+    SNAPSHOT: str = "snapshot"
+    TARGETS: str = "targets"
+    TIMESTAMP: str = "timestamp"
+
+    @staticmethod
+    def all() -> set:
+        return set(map(lambda r: r.value, RoleNames))
```

Comment:
```suggestion
class RoleNames(str, Enum):
    ROOT: str = "root"
    SNAPSHOT: str = "snapshot"
    TARGETS: str = "targets"
    TIMESTAMP: str = "timestamp"

    @staticmethod
    def all() -> set:
        return set(map(lambda r: r.value, RoleNames))
```

If you inherit the string class as well you will be able to do comparisons like:
```
if RoleNames.TARGETS == role
```
instead of doing:
```
if RoleNames.TARGETS.value == role
```
Other such examples from python docs:
https://docs.python.org/3/library/enum.html#others

Reply:
Additionally, when do we use `RoleNames.all()`? Why do we need it?

Reply:
It refers to @sechkova's comment for removing `TOP_LEVEL_ROLE_NAMES`. Enums in python don't have any not-breaking-the-rules option to access all their values in a list/set/etc.

Reply:
I'm fine to go with simple constants if that's the agreement, even though in perspective it doesn't feel like the best option. For example, one would need to always use 
```
from tuf.api.metadata import  Root, ROOT, Targets, TARGETS, Snapshot, SNAPSHOT, Timestamp, TIMESTAMP
```
instead of
```
from tuf.api.metadata import  Root, Targets, Snapshot, Timestamp, RoleNames
```

and if for some reason in future let's say timestamp for example is used in another context as well within TUF, using only TIMESTAMP constant might be confusing.

I changed to enum to address @sechkova's proposal, but if the agreement is to use constants, then we can keep `TOP_LEVEL_ROLE_NAMES `.

Reply:
> For example, one would need to always use
> 
> from tuf.api.metadata import  Root, ROOT, Targets, TARGETS, Snapshot, SNAPSHOT, Timestamp, TIMESTAMP

This is why I've never done this*... it seems very hard to provide this as API that is not a repetitive mess but still offers the role names in a way that allows elegant code.

 *) I'm very happy someone is trying though: there could be a neat solution we haven't seen yet